src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { DispatchContext } from '../../utils/GlobalContext';
+
+type Props = {
+  todo: Todo;
+};
+type SubmitProps =
+  | React.FocusEvent<HTMLInputElement>
+  | React.KeyboardEvent<HTMLInputElement>
+  | React.FormEvent<HTMLFormElement>;
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const dispatch = useContext(DispatchContext);
+
+  const { title, completed } = todo;
+  const [editing, setEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const inputElem = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputElem.current) {
+      inputElem.current.focus();
+    }
+  }, [editing]);
+
+  const handleCancel = () => {
+    setEditing(false);
+    setNewTitle(todo.title);
+  };
+
+  const handleKeyEvent = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.type === 'keyup') {
+      const keyEvent = event as React.KeyboardEvent<HTMLInputElement>;
+
+      if (keyEvent.key === 'Escape') {
+        handleCancel();
+      }
+    }
+  };
+
+  function handleSubmit(event: SubmitProps): void {
+    event.preventDefault();
+    setEditing(false);
+
+    if (newTitle.length === 0) {
```

Comment:
```suggestion
    if (!newTitle.length) {
```
shorter form
Alos, trim value to avoid adding spaces only